corehq/apps/app_manager/views/download.py

```diff
@@ -206,9 +208,13 @@ def download_file(request, domain, app_id, path):
         content_type = None
     response = HttpResponse(content_type=content_type)
 
+    build_profile = request.GET.get('profile')
+    build_profile_access = domain_has_privilege(domain, privileges.BUILD_PROFILES)
     if path in ('CommCare.jad', 'CommCare.jar'):
         set_file_download(response, path)
         full_path = path
+    elif build_profile and build_profile in request.app.build_profiles and build_profile_access:
```

Comment:
Is it true that if a user tries to download a specific build profile by specifying `profile=<profile_id>` in the URL, but that profile ID does not exist in the `app.build_profiles` then this will silently ignore the specified profile and serve up the default build? Would it make sense to raise a Http404 here instead? As is it will be hard for us to tell if we ever had a bug that put the wrong profile ID in a URL.